### PR TITLE
shutdown quicker on db server and coordinator heartbeat threads

### PR DESCRIPTION
**Pull request by matthewvon**
Ignore lines 210 to 212 in HeartbeatThread.cpp.  They are for the next PR and have no impact here.  Same with line 942.

Recommend adding ?w=1 to URL when viewing file changes.

### CODE REVIEW

File: arangod/Cluster/HeartbeatThread.cpp [hunk 1]
```diff
@@ -595,14 +605,15 @@ void HeartbeatThread::runSingleServer() {
         config._endpoint = endpoint;
         config._autoResync = true;
         config._autoResyncRetries = 2;
+        LOG_TOPIC(INFO, Logger::HEARTBEAT) << "start initial sync from leader";
```
Comment (graetzer) [inline arangod/Cluster/HeartbeatThread.cpp]: I think the log message is no longer in devel ?

File: lib/Basics/ConditionVariable.h [hunk 2]
```diff
@@ -99,10 +101,11 @@ class ConditionVariable {
 
   //////////////////////////////////////////////////////////////////////////////
   /// @brief waits for an event with timeout in micro seconds
-  /// returns true when the condition was signaled, false on timeout 
+  /// returns true when the condition was signaled, false on timeout
   //////////////////////////////////////////////////////////////////////////////
 
   bool wait(uint64_t);
+  bool wait(std::chrono::microseconds);
```
Comment (graetzer) [inline lib/Basics/ConditionVariable.h]: If your're adding this, maybe change this line to use `std::chrono` directly too: https://github.com/arangodb/arangodb/blob/ed94894103c171267b8dce4a9fb578afe918c46e/lib/Basics/ConditionLocker.cpp#L86

Comment (matthewvon) [reply lib/Basics/ConditionVariable.h]: @neunhoef Looking to you for merge.

Comment (kvahed) [reply lib/Basics/ConditionVariable.h]: OK. But keep in mind that duration</*signed integer type of at least 55 bits*/, std::micro> so 55bits only. if you need 64 bits, do not use microseconds.

**Pull request merged**